test/specs/modules/Dropdown/Dropdown.spec.js

```diff
@@ -54,4 +113,217 @@ describe('Dropdown', () => {
+    expect(wrapper.emitted('input')[2][0]).to.deep.equal([2]);
+  });
+
+  it('should not select more than max-selections', () => {
+    const wrapper = shallow(DropdownWithRequired, {
+      propsData: {
+        multiple: true,
+        maxSelections: 1,
+        options: [{ text: 'foo', value: 1 }, { text: 'bar', value: 2 }],
+      },
+    });
+
+    wrapper.vm.$on('input', (value) => {
+      wrapper.setProps({ value });
+    });
+
+    const options = wrapper.findAll(DropdownItem);
+
+    options.at(0).trigger('click');
+    expect(wrapper.emitted('input')[0][0]).to.deep.equal([1]);
+    options.at(0).trigger('click');
+    expect(wrapper.emitted('input').length).to.equal(1);
+  });
+
+  it('should have icons, flags and images in selected text', () => {
+    const wrapper = shallow(DropdownWithRequired, {
+      propsData: {
+        options: [
+          {
+            text: 'foo',
+            value: 1,
+            flag: 'cn',
+            icon: 'question',
+            image: {
+              src: '/test',
+            },
+          },
+        ],
+      },
+    });
+    wrapper.vm.$on('input', (value) => {
+      wrapper.setProps({ value });
+    });
+
+    wrapper.find(DropdownItem).trigger('click');
+
+    const text = wrapper.find('div.text');
+
+    const icon = text.find(Icon);
+    expect(icon.is(Icon)).to.equal(true);
+    const flag = text.find(Flag);
+    expect(flag.is(Flag)).to.equal(true);
+    const image = text.find(Image);
+    expect(image.is(Image)).to.equal(true);
+  });
+
+  it('should have icons, flags and images in selected options', () => {
+    const wrapper = shallow(DropdownWithRequired, {
+      propsData: {
+        multiple: true,
+        options: [
+          {
+            text: 'foo',
+            value: 1,
+            flag: 'cn',
+            icon: 'question',
+            image: {
+              src: '/test',
+            },
+          },
+        ],
+      },
+    });
+    wrapper.vm.$on('input', (value) => {
+      wrapper.setProps({ value });
+    });
+
+    wrapper.find(DropdownItem).trigger('click');
+
+    const label = wrapper.find(Label);
+
+    const icon = label.find(Icon);
+    expect(icon.is(Icon)).to.equal(true);
+    const flag = label.find(Flag);
+    expect(flag.is(Flag)).to.equal(true);
+    const image = label.find(Image);
+    expect(image.is(Image)).to.equal(true);
+  });
+
+  it('should filter options', () => {
+    const wrapper = shallow(DropdownWithRequired, {
+      propsData: {
+        search: true,
+        options: [{ text: 'foo' }, { text: 'bar' }, { text: 'baz' }],
+      },
+    });
+    const input = wrapper.find('input.search');
+    input.element.value = 'ba';
+    input.trigger('input');
+    expect(wrapper.vm.filteredOptions).to.deep.equal([{ text: 'bar' }, { text: 'baz' }]);
+    input.element.value = 'blah';
+    input.trigger('input');
+    expect(wrapper.vm.filteredOptions).to.deep.equal([]);
+    expect(wrapper.vm.message).to.equal('No results found');
```

Comment:
Same as above: `vm` is private and reflects an internal behavior. Instead, unit tests should only consider the output (ie: html, classes, events).